### ISSUE TITLE
Bot not skipping immunes?

### ISSUE THREAD

**Issue by ricotta3d** [kolton/d2bot-with-kolbot#810]
I set my bot to skip cold immunes but it's still attacking them with lightning (which I set up as a second skill to use vs immune monsters, but only did this because the code requires this jargon it seems).

Is this working as intended?

**Comment by DarkHorseDre** [comment]
I cant check now but what you say makes sense: if you have `Config.AttackSkill[5]` set bot will use lightning to attack cold immune, even if you have `Config.SkipImmune = [cold];` set.  just dont set `Config.AttackSkill[5]` in this case, and cold immunes should be skipped.

**Comment by ricotta3d** [comment]
That makes sense. I didn't think to simply not set the other skill and use // to omit the field. Thank you

**Comment by ricotta3d** [comment]
How would I go about not setting it, or rather what's the jargon for the code? Thanks for your help!

**Comment by DarkHorseDre** [comment]
yes -1 disables (as default) dont comment out array lines like that - will break stuff as the code needs to find them, even if the values are disabled (hence the -1 option)

**Comment by ricotta3d** [comment]
hmmm, it seems the -1 option to disable still results in my bot teleporting around the map trying to kill cold immunes?

**Comment by DarkHorseDre** [comment]
ok that shouldn't happen. please post a link to a paste in of your config, and any other files (e. g. attack.js) you have modified

**Comment by ricotta3d** [comment]
Sure, here it is!

https://pastebin.com/LNcBJHg4

**Comment by DarkHorseDre** [comment]
Hmmm I cant see an issue with your skill or immune setup - does it do this with every cold immune or mainly in one/certain bot scripts?
Does it happen only with cold immunes or dual/triple immunes? (e.g. cold and fire).
Have you modified any other scripts? (attack.js, bot scripts etc)

if it is any/all cold immunes (including solo cold immunes) I would download a new copy of bot, have it generate a new char config for existing toon (don't copy it), reconfigure and see if it happens, because it shouldn't do..

**Comment by ricotta3d** [comment]
I'll try that then, thanks.

I haven't modified any other scripts (did modify Chaos Sanctuary script, but I haven't been running this).
Happens to cold cold + dual/triple immunes.
I've seen this happen in both Trav + Throne Room.

**Comment by ricotta3d** [comment]
After disabling lightning, it does not attempt to use lightning. Instead, it just teleports around casting blizzard. Console doesn't say anything special when this happens.

Will report back if downloading fixes. For the time being, I've set a max timer on the games.

**Comment by DarkHorseDre** [comment]
one screwy thing I saw is youre using blizzard (a timed skill) as pre-attack.
`Config.AttackSkill[0] = 59; // Preattack skill.`

disable that (set to -1) and see what happens.
